src/strings.cpp

```diff
@@ -1180,75 +1172,75 @@ static void FormatString(StringBuilder &builder, std::string_view str_arg, Strin
 						}
 
 						/* The gender is stored at the start of the formatted string. */
-						const char *s = buffer.c_str();
-						char32_t c = Utf8Consume(&s);
-						/* Does this string have a gender, if so, set it */
-						if (c == SCC_GENDER_INDEX) gender = (uint8_t)s[0];
+						{
+							StringConsumer tmp_consumer(buffer);
+							/* Does this string have a gender, if so, set it */
+							if (tmp_consumer.ReadUtf8If(SCC_GENDER_INDEX)) {
+								gender = tmp_consumer.ReadUint8();
+							}
+						}
```

Comment:
The extra scoping isn't really needed, and maybe use 'gender_consumer' as variable name?